### PR TITLE
Remove syncthing process based on name

### PR DESCRIPTION
**Pull request by pchico83**
Using `p.Name` instead of `p.Exe` solves the cross-compilation issues we have seen before

### CODE REVIEW

File: pkg/syncthing/syncthing.go [hunk 1]
```diff
@@ -701,33 +670,51 @@ func (s *Syncthing) Restart(ctx context.Context) error {
...
+			log.Infof("error terminating syncthing %d with wait: %s", p.Pid, err.Error())
 		}
+		log.Infof("terminated syncthing %d with wait: %s", p.Pid, s.Home)
 	}
+	return nil
+}
 
-	if err := s.cleanupDaemon(pid, force); err != nil {
-		return err
+// SoftTerminate best effor to halt the background process
```
Comment (rberrelleza) [inline pkg/syncthing/syncthing.go:704]: This doesn't explain what the function does. Try to rephrase it?

File: pkg/syncthing/syncthing.go [hunk 2]
```diff
@@ -701,33 +670,51 @@ func (s *Syncthing) Restart(ctx context.Context) error {
...
+func (s *Syncthing) SoftTerminate() error {
```
Comment (rberrelleza) [inline pkg/syncthing/syncthing.go:705]: Both soft and hard Terminate kill the process using the terminate function. What makes one soft vs hard?

**Review by rberrelleza — COMMENTED** [review]
Tests?

File: pkg/syncthing/process.go [hunk 3]
```diff
@@ -16,22 +14,30 @@
...
+	if !wait {
+		return nil
+	}
+
+	isRunning, err := p.IsRunning()
+	if err != nil {
+		return err
+	}
+	for isRunning {
+		time.Sleep(10 * time.Millisecond)
```
[the remark this answers is not in the document]
Comment (pchico83) [reply pkg/syncthing/process.go:36]: Oh, good catch. I will also kill it if it takes long

**Pull request merged**